runtime-tools-quarkus-extension-parent/runtime/src/main/java/org/kie/kogito/runtime/tools/quarkus/extension/runtime/forms/impl/FormsStorageImpl.java

```diff
@@ -37,39 +54,120 @@
-                .collect(Collectors.toList());
+    public Form getFormContent(String formName) throws IOException {
+        File formFile = getFormFile(formName);
+        File formConfig = getFormConfigFile(formName);
+        FormConfiguration formConfiguration = null;
+        if (formConfig != null && formConfig.exists()) {
+            formConfiguration = new FormConfiguration(IOUtils.toString(new FileInputStream(formConfig), Charset.forName("UTF-8")));
+        } else {
+            throw new FileNotFoundException("{0} dose not found");
+        }
+        Form form = new Form(formConfiguration);
+        if (formFile != null && formFile.exists()) {
+            form.setSource(IOUtils.toString(new FileInputStream(formFile), Charset.forName("UTF-8")));
+        } else {
+            throw new FileNotFoundException("{0}'s config file dose not found");
+        }
+        return form;
     }
 
-    private Collection<File> readForms() {
-        File rootFolder = FileUtils.toFile(formsFolder);
+    private File getFormFile(String formName) {
+        Collection<FormInfo> formBaseInfo = formBaseInfoCollection.get(formName);
+        if (formBaseInfo != null) {
+            Optional<FormInfo> optional = formBaseInfo.stream().filter(formBase -> !formBase.getType().getValue().equals("config")).findFirst();
 
-        return FileUtils.listFiles(rootFolder, new String[] { "html", "tsx" }, false);
+            if (optional.isPresent()) {
+                return optional.get().getFile();
+            }
+        }
+        return null;
+    }
 
+    private File getFormConfigFile(String formName) {
+        Collection<FormInfo> formBaseInfo = formBaseInfoCollection.get(formName);
+        if (formBaseInfo != null) {
+            Optional<FormInfo> optional = formBaseInfo.stream().filter(formBase -> formBase.getType().getValue().equals("config")).findFirst();
+
+            if (optional.isPresent()) {
+                return optional.get().getFile();
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public void updateFormContent(String formContent, String formName) throws IOException {
+        Form form = new Form(formContent);
+        File formFile = getFormFile(formName);
+        IOUtils.write(form.getSource(), new FileOutputStream(formFile), Charset.forName("UTF-8"));
+        File formConfigFile = getFormConfigFile(formName);
+        IOUtils.write(form.getConfiguration().toString(), new FileOutputStream(formConfigFile), Charset.forName("UTF-8"));
+    }
+
+    @Override
+    public void reloadFormBaseInfoList() {
+        formBaseInfoCollection = readFormBaseList().stream()
+                .map(file -> new FormInfo(FilenameUtils.removeExtension(file.getName()), getFormType(FilenameUtils.getExtension(file.getName())), new Date(file.lastModified()), file))
```

Comment:
We should have a FormInfo for each Form file (TSX or HTML) not for the config. We can assume there will be only one form being tsx or html, not both of them